providers/openshift/provider.go

```diff
@@ -131,18 +138,29 @@ func (p *OpenShiftProvider) newOpenShiftClient() (*http.Client, error) {
 		capaths = paths
 		system_roots = false
 	}
+
+	// try to retrieve a cached client
+	metadataHash, err := util.GetFilesMetadataHash(capaths)
```

Comment:
This does not handle if system roots change (thinking about the proxy CA work).

Reply:
Yeah, I was aware of that when I wrote the code, we currently use the Go methods that read the system roots only once.
I think the particular Go-crypto implementation that actually tries to read one of the system roots possible locations is unimported, so an option is to go platform-dependent and hardcode system roots path. Given the main oauth-proxy use-case is to be a side-container, and we can control the system it's built atop, that hard-code might be ok, although it can break some assumtions, like respecting `SSL_CERT_FILE`/`SSL_CERT_DIR` env vars or people who might try to build the proxy on their own non-Red Hat system...